### DISCUSSION TITLE
BeforeMap not called with open generic

### DISCUSSION THREAD

**Discussion by sebastiencozzi** [LuckyPennySoftware/AutoMapper#4047] [NA]
### Source/destination types

CustomList to List

### Mapping configuration

```csharp
            cfg.CreateMap(typeof(ModifiableList<>), typeof(List<>))
                    .BeforeMap((a, b) =>//Never called
                    {
                        Type t = a.GetType();
                        if(t == typeof(int))
                        {
                            throw new InvalidOperationException("A is an int");
                        }
                        else
                        {
                            throw new InvalidOperationException("A isn't an int");
                        }
                        return;
                })
                    .ConvertUsing(typeof(ModifiableListConverter<,>))
                    ;
```

### Version: 11.0.1
   .NET 6 with automapper 11.0.1

### Expected behavior
The BeforeMap callback is called before the TypeConverter

### Actual behavior
The BeforeMap callback is not called

### Steps to reproduce

https://gist.github.com/sebastiencozzi/b4a667d0965d422c7689d3cb855758d8#file-beforemapopengeneric

**Comment by lbargaoanu** [comment]
That's expected. Type converters don't mix with other configuration. You're, in fact, taking over the mapping.
